File: Efficiently working with data readers for SQL-Server in VB.NET/[VB]-Efficiently working with data readers for SQL-Server in VB.NET/VB/TeamBaseLibrary/BaseSqlServerConnections.vb
Imports ExceptionsLibrary
''' <summary>
''' Pre-configure connection string using a default server 
''' and catalog. In a normal solution this is all that should
''' be required but if at run time the connection needs to
''' change you can change the server and catalog in the class
''' which inherits from this class.
''' 
''' BaseExceptionsHandler provides properties which any
''' class inherits from BaseSqlServerConnections can set
''' a property then when the caller method throws an exception
''' you can mark a property that an exception occured
''' then in the high level caller (e.g. in a form) can
''' ask "was there an exception" and if so get the exception
''' via LastException property in BaseExceptionsHandler.
''' </summary>
Public Class BaseSqlServerConnections
    Inherits BaseExceptionsHandler

    ''' <summary>
    ''' This points to your database server
    ''' </summary>
    Protected DatabaseServer As String = "KARENS-PC"
    ''' <summary>
    ''' Name of database containing required tables
    ''' </summary>
    Protected DefaultCatalog As String = ""
    Public ReadOnly Property ConnectionString As String
        Get
            Return $"Data Source={DatabaseServer};" &
                   $"Initial Catalog={DefaultCatalog};Integrated Security=True"
        End Get
    End Property


End Class

File: Export Excel from SQL-Server/[C#][VB]-Export Excel from SQL-Server/C# and VB/Operations_cs/BaseSqlServerConnections.cs
using System;

namespace Operations_cs
{
    public class BaseSqlServerConnections
    {
        /// <summary> 
        /// This points to your database server 
        /// </summary> 
        protected string DatabaseServer = "KARENS-PC";
        /// <summary> 
        /// Name of database containing required tables 
        /// </summary> 
        protected string DefaultCatalog = "";
        public string ConnectionString => $"Data Source={DatabaseServer};Initial Catalog={DefaultCatalog};Integrated Security=True";

        /// <summary> 
        /// Determines if running on Karen Payne's computer 
        /// </summary> 
        public bool IsKarenMachine => Environment.UserName == "Karens";
        /// <summary> 
        /// Determine if server name has been set from the default on Karen Payne's computer 
        /// </summary> 
        public bool IsKarensDatabaseServer => DatabaseServer == "KARENS-PC";
    }
}
